cmake/modules/hunter_add_repository.cmake

```diff
@@ -0,0 +1,81 @@
+include(CMakeParseArguments) # cmake_parse_arguments
+include(hunter_internal_error)
+include(hunter_fatal_error)
+include(hunter_test_string_not_empty)
+include(hunter_status_debug)
+include(hunter_status_print)
+include(hunter_install_repository)
+
+function(hunter_add_repository)
+  if(PROJECT_NAME OR NOT HUNTER_GATE_ROOT)
```

Comment:
> I haven't worked with sub-projects in the past, so I'm not sure I understand the pros and cons completely

This is a trivial way to add new stuff to local build. It's not a correct way in long terms but fast, good for experimenting.
So you have project with library `foo`:
```
cmake_minimum_required(VERSION 3.0)
project(Foo)
add_library(foo foo.cpp)
```

And you have project with executable `boo` and you want to use `foo`:
```
cmake_minimum_required(VERSION 3.0)
project(Boo)

add_subdirectory(/path/to/.../Foo) # or add_subdirectory(Foo) if it's local

add_executable(boo boo.cpp)
target_link_libraries(boo PUBLIC foo) # use foo from Foo project
```